docs/upgrade-to-v3.md

```diff
@@ -33,6 +33,136 @@ The Microsoft Graph Java SDK now leverages azure-identity to improve the authent
 1. Remove any dependency on `microsoft-graph-android-auth` and replace them by one of the providers available in azure-identity.
 1. Reconsider the need for custom authentication providers (implementation of `IAuthenticationProvider` or `ICoreAuthenticationProvider`) that might be part of your project and use a provider available in azure-identity whenever possible.
 
+#### Authentication upgrade examples
+
+The purpose of this section is to provide examples on how to upgrade your authentication providers to azure-identity providers.
+
+##### Authorization code flow
+
+Replace any of the following:
+
+```Java
+AuthorizationCodeProvider authProvider = new AuthorizationCodeProvider(CLIENT_ID, SCOPES, AUTHORIZATION_CODE, REDIRECT_URL, NATIONAL_CLOUD, TENANT, CLIENT_SECRET);
+IGraphServiceClient graphClient = GraphServiceClient.builder().authenticationProvider(authProvider).buildClient();
+```
+
+By
+
+``` Java
+final AuthorizationCodeCredential authCodeCredential = new AuthorizationCodeCredentialBuilder()
+    .clientId(CLIENT_ID)
+    .clientSecret(CLIENT_SECRET) //required for web apps, do not set for native apps
+    .authorizationCode(AUTHORIZATION_CODE)
+    .redirectUrl(REDIRECT_URL)
+    .build();
+final TokenCredentialAuthProvider tokenCredentialAuthProvider = new TokenCredentialAuthProvider(SCOPES, authCodeCredential);
+final GraphServiceClient graphClient = GraphServiceClient.builder().authenticationProvider(authProvider).buildClient();
```

Comment:
Shouldn't it be "tokenCredentialAuthProvider"?

Reply:
thanks, fixed.